Libraries/LibWeb/DOM/NamedNodeMap.cpp

```diff
@@ -148,13 +148,12 @@ Attr const* NamedNodeMap::get_attribute(FlyString const& qualified_name, size_t*
         *item_index = 0;
 
     // 1. If element is in the HTML namespace and its node document is an HTML document, then set qualifiedName to qualifiedName in ASCII lowercase.
-    // FIXME: Handle the second condition, assume it is an HTML document for now.
-    bool compare_as_lowercase = associated_element().namespace_uri() == Namespace::HTML;
+    bool compare_as_lowercase = associated_element().namespace_uri() == Namespace::HTML && associated_element().document().is_html_document();
 
     // 2. Return the first attribute in element’s attribute list whose qualified name is qualifiedName; otherwise null.
     for (auto const& attribute : m_attributes) {
         if (compare_as_lowercase) {
-            if (attribute->name().equals_ignoring_ascii_case(qualified_name))
+            if (!AK::any_of(attribute->name().bytes(), is_ascii_upper_alpha) && attribute->name().equals_ignoring_ascii_case(qualified_name))
                 return attribute;
         } else {
             if (attribute->name() == qualified_name)
```

Comment:
it appears that it’s possible now to use the guarantee of the first clause in the second one…? (further contractions are seemingly possible too.) 🤔

Reply:
It's definitely possible to simplify the `!AK::any_of(attribute->name().bytes(), is_ascii_upper_alpha) && attribute->name().equals_ignoring_ascii_case(qualified_name)` condition so that the strings are only iterated once. Is that what you're suggesting? I can't see anything other than that that wouldn't be trivial for the compiler to optimize. Is there a specific change you had in mind?

Reply:
i thought that `!AK::any_of(attribute->name().bytes(), is_ascii_upper_alpha) && attribute->name() == qualified_name` might be possible ( and if so, then the second clause repeats below).

Reply:
>  thought that !AK::any_of(attribute->name().bytes(), is_ascii_upper_alpha) && attribute->name() == qualified_name might be possible ( and if so, then the second clause repeats below).

Ah, I see. No, you can't do that because the second part of the check would fail if any part of `qualified_name` was uppercase.